File: CareConnect/CareConnect.Models/Requests/EnrollmentRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace CareConnect.Models.Requests
{
    public class EnrollmentRequest
    {
        public int ClientId { get; set; }

        public int? ChildId { get; set; }

        public int WorkshopId { get; set; }

        public string? PaymentIntentId { get; set; } // null for free workshops
    }
}

File: UI/careconnect_mobile/lib/models/requests/enrollment_request.dart
import 'package:careconnect_mobile/models/requests/appointment_insert_request.dart';

import 'package:json_annotation/json_annotation.dart';

part 'enrollment_request.g.dart';

@JsonSerializable()
class EnrollmentRequest {
  int clientId;
  int? childId;
  int workshopId;
  String? paymentIntentId;

  EnrollmentRequest({
    required this.clientId,
    required this.childId,
    required this.workshopId,
    required this.paymentIntentId,
  });

  factory EnrollmentRequest.fromJson(Map<String, dynamic> json) =>
      _$EnrollmentRequestFromJson(json);

  Map<String, dynamic> toJson() => _$EnrollmentRequestToJson(this);
}
